### PR TITLE
feat: [EXC-1675] migrate replica to no-op LogVisibilityV2

### PR DESCRIPTION
**Pull request by maksymar**
This PR migrates replica to using a no-op `LogVisibilityV2` instead of `LogVisibility`.

The difference between the two is in `AllowedViewers` enum variant which is currently ignored, so it's expected for V2 to behave the same way as V1.

### CODE REVIEW

**Comment by maksymar** [comment]
Note: do not submit before updating nns/sns proto messages.

**Comment by dsarlis** [comment]
> Note: do not submit before updating nns/sns proto messages.

I think we can let the NNS team take care of updating their interface if they are interested to use the new log visibility and focus only on updating the replica to use the new type (canisters can still send the old type and we know what to do with it). There's no reason to add a dependency in our process on the NNS/SNS canisters to be upgraded before we can proceed with further steps.

**Comment by maksymar** [comment]
> > Note: do not submit before updating nns/sns proto messages.
> 
> I think we can let the NNS team take care of updating their interface if they are interested to use the new log visibility and focus only on updating the replica to use the new type (canisters can still send the old type and we know what to do with it). There's no reason to add a dependency in our process on the NNS/SNS canisters to be upgraded before we can proceed with further steps.

I had to update `rs/nns/cmc/cmc.did` with new type so that their test passes and after that I suppose we have to update nns proto.

**Comment by dsarlis** [comment]
> > > Note: do not submit before updating nns/sns proto messages.
> > 
> > 
> > I think we can let the NNS team take care of updating their interface if they are interested to use the new log visibility and focus only on updating the replica to use the new type (canisters can still send the old type and we know what to do with it). There's no reason to add a dependency in our process on the NNS/SNS canisters to be upgraded before we can proceed with further steps.
> 
> I had to update `rs/nns/cmc/cmc.did` with new type so that their test passes and after that I suppose we have to update nns proto.

What if you don't make the change in their tests to use V2? I doubt that you'd need to change any interface in that case. We can leave this update to after the replica fully uses V2 (the only thing blocked is completely removing V1 which is something that can wait a bit longer).

File: rs/nns/cmc/cmc.did
Comment (aterga) [inline rs/nns/cmc/cmc.did:6]: Can we make this optional?

```suggestion
  allowed_viewers : opt vec principal;
```

**Review by aterga — APPROVED** [review]
LGTM modulo CMC field optionality.

**Review by schneiderstefan — APPROVED** [review]

**Comment by maksymar** [comment]
> LGTM modulo CMC field optionality.

I removed cmc.did change, and added a copy-type to overcome type dependency from management canister. PTAL.

File: rs/nns/integration_tests/src/cycles_minting_canister.rs
Comment (aterga) [inline rs/nns/integration_tests/src/cycles_minting_canister.rs:688]:
```suggestion
        settings: settings.map(CanisterSettingsArgs::from),
```

File: rs/nns/integration_tests/src/cycles_minting_canister.rs
Comment (aterga) [inline rs/nns/integration_tests/src/cycles_minting_canister.rs:775]:
```suggestion
        settings: settings.map(CanisterSettingsArgs::from),
```

File: rs/nns/cmc/src/lib.rs
Comment (aterga) [inline rs/nns/cmc/src/lib.rs:86]: Is this intended to stay? If so, please add a code comment to explain why.

**Review by aterga — APPROVED** [review]

File: rs/execution_environment/src/query_handler.rs
Comment (dsarlis) [inline rs/execution_environment/src/query_handler.rs:287]: Not a strong opinion: I'm not sure the flag adds a lot of value. We could probably just unconditionally map `AllowedViewers` to default for now and then releasing the feature would simply be removing that code and allowing the field to go through as provided.

File: rs/nns/integration_tests/src/cycles_minting_canister.rs
Comment (maksymar) [reply rs/nns/integration_tests/src/cycles_minting_canister.rs]: Done.

File: rs/nns/integration_tests/src/cycles_minting_canister.rs
Comment (maksymar) [reply rs/nns/integration_tests/src/cycles_minting_canister.rs]: Done.

**Review by aterga — APPROVED** [review]

**Review by dsarlis — APPROVED** [review]

File: rs/execution_environment/src/query_handler.rs
Comment (maksymar) [reply rs/execution_environment/src/query_handler.rs:287]: It's easier for me just to flip this flag and see what tests fail and why. Because of that I put a corresponding comment to the specific test that has to be changed accordingly.

**Pull request merged**
